localtypings/pxtarget.d.ts

```diff
@@ -40,13 +40,27 @@ declare namespace pxt {
         bannedRepos?: string[];
         allowUnapproved?: boolean;
         preferredRepos?: string[]; // list of company/project(#tag) of packages to show by default in search
+        preferredRepoLib?: RepoData[];
+        approvedRepoLib?: RepoData[];
         // format:
         // "acme-corp/pxt-widget": "min:v0.1.2" - auto-upgrade to that version
         // "acme-corp/pxt-widget": "dv:foo,bar" - add "disablesVariant": ["foo", "bar"] to pxt.json
         upgrades?: pxt.Map<string>;
         // list of trusted custom editor extension urls
         // that can bypass consent and send/receive messages
         approvedEditorExtensionUrls?: string[];
+        extensionsToIgnore?: string[]; // List of extensions to ignore when allowing for deletion
+        categories?: ExtensionCategory[];
+    }
+
+    interface RepoData {
+        slug: string;
+        tags: string[]
```

Comment:
This is pretty minor, but looking at the other pr there are a lot with empty tags arrays, it might be nice to set this as optional field / make sure it's guarded against it being undefined wherever it's referenced -- targetconfig is loaded live and updated regularly so it can help minimize future mistakes by minimizing the amount of things that are required to be set.

Reply:
ah yeah i'll change it!